File: backend/src/main/java/com/manabihub/writing/service/WritingService.java
package com.manabihub.writing.service;

import com.manabihub.writing.dto.request.SubmitWritingRequest;
import com.manabihub.writing.dto.response.WritingAssignmentResponse;
import com.manabihub.writing.dto.response.WritingResultResponse;

import java.util.UUID;

public interface WritingService {

    WritingAssignmentResponse getAssignment(UUID lessonBlockId);

    WritingResultResponse submit(SubmitWritingRequest request);

}

File: frontend/src/features/writing/services/writingService.ts
import { axiosClient } from "../../../shared/api/axiosClient";
import { ENDPOINTS } from "../../../shared/api/endpoints";

import type {
    SubmitWritingRequest,
    WritingAssignmentResponse,
    WritingResultResponse,
} from "../types";

export const writingService = {

    getAssignment: async (
        lessonBlockId: string
    ): Promise<WritingAssignmentResponse> => {

        const res = await axiosClient.get(
            ENDPOINTS.studentWriting.assignment(lessonBlockId)
        );

        return res.data.data;
    },

    submitWriting: async (
        payload: SubmitWritingRequest
    ): Promise<WritingResultResponse> => {

        const res = await axiosClient.post(
            ENDPOINTS.studentWriting.submit,
            payload
        );

        return res.data.data;
    },
};